### PR TITLE
try to fix comment bad wrapping

### PR DESCRIPTION
**Pull request by d-dorazio**
This an attempt to solve #3059 . I _think_ the problem was that `rewrite_string` was using `shape.max_chars_with_indent` to reset the maximum chars per line, while there could be more possibly. This PR adds a new parameter that is the number of characters per line and use that when `break_string` returns `LineEnd`.

It's my first pr, so I might be completely wrong :)

### CODE REVIEW

**Comment by d-dorazio** [comment]
the integration with `stdsimd` is failing, but I think that `rustfmt` has not been run on it because I tried to run `rustfmt` without these changes on it and there were a lot of changes.

File: src/comment.rs [hunk 1]
```diff
@@ -579,7 +579,11 @@ fn rewrite_comment_inner(
             let item_fmt = ib.create_string_format(&fmt);
             result.push_str(&comment_line_separator);
             result.push_str(&ib.opener);
-            match rewrite_string(&item_block_buffer.replace("\n", " "), &item_fmt) {
+            match rewrite_string(
+                &item_block_buffer.replace("\n", " "),
+                &item_fmt,
+                max_chars - ib.indent,
```
Comment (topecongiro) [inline src/comment.rs]: Would you use `saturating_sub` here (and in other places as well) to avoid panicking on extreme corner cases please?

Comment (d-dorazio) [reply src/comment.rs]: makes sense, thanks!

**Comment by otavio** [comment]
I am a new contributor here so I just want to give my 2c. Personally, I dislike to have multiple commits which address same issue, if they are logically connected.

So looking at this changes, I would:

- squash both commits
- reword the commit so it includes the explanation you did on the PR

Later, looking at the commit logs, it provide a nice reasoning of the changes. Also having "atomic" changes makes much easier to use `git bisect` to find out the reason of regressions.

**Comment by d-dorazio** [comment]
@otavio I agree for the most part, but I prefer to squash only when merging the pr and not while working on it. This way reviewers can easily see what changed from last time they looked at the code without wasting time reading code that hasn't changed. 

I'm not sure if there are any policies about this in this repo, but I think that squashing this PR should be fine.

**Comment by topecongiro** [comment]
Thank you for the update!

**Comment by ayazhafiz** [comment]
Closes #3059

**Pull request merged**